### PR TITLE
ADO132072 - new java images

### PR DESCRIPTION
**Pull request by AlessandroDeMartini**

### CODE REVIEW

**Review by aielloaldo — APPROVED** [review]

**Review by luapicella — APPROVED** [review]

File: openjdk/17/debian/create-image-from-git-tag.Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            }
+        }
+        stage('Remove Unused docker image') {
+            steps{
+                sh "docker rmi ${imageName}:${params.TAG}"
+            }
+        }
+    }
+    post {
+        failure {
```
Comment (agentili) [inline openjdk/17/debian/create-image-from-git-tag.Jenkinsfile:91]: Is this failure step actually needed. 
"env.BRANCH_NAME" I don't know if it is present as a variable

File: openjdk/17/debian/create-image-from-git-tag.Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            emailext( attachLog: true,
+                compressLog: true,
+                recipientProviders: [culprits()],
+                subject: "[${currentBuild.projectName}] Failed Pipeline: ${currentBuild.fullDisplayName}",
+                body: "Something is wrong with ${env.BUILD_URL} on branch ${env.BRANCH_NAME}, please check the log"
+            )
+        }
+        always {
+            updateGithubCommitStatus("Jenkins Job ${env.BUILD_NUMBER} - Result: ${currentBuild.currentResult} ")
```
Comment (agentili) [inline openjdk/17/debian/create-image-from-git-tag.Jenkinsfile:100]: is it actually usefull here when we build from TAG?

File: openjdk/17/debian/create-image-from-git-tag.Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    sh "git config --get remote.origin.url > .git/remote-url"
+    return readFile(".git/remote-url").trim()
+}
+
+def getCommitSha() {
+    sh "git rev-parse HEAD > .git/current-commit"
+    return readFile(".git/current-commit").trim()
+}
+
+def updateGithubCommitStatus(description) {
```
Comment (agentili) [inline openjdk/17/debian/create-image-from-git-tag.Jenkinsfile:14]: gitRepoURL, getCommitSha and updateGithubCommitStatus I think are not needed here

File: maven/3.8.7/openjdk17/create-image-from-git-tag.Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    sh "git config --get remote.origin.url > .git/remote-url"
+    return readFile(".git/remote-url").trim()
+}
+
+def getCommitSha() {
+    sh "git rev-parse HEAD > .git/current-commit"
+    return readFile(".git/current-commit").trim()
+}
+
+def updateGithubCommitStatus(description) {
```
Comment (agentili) [inline maven/3.8.7/openjdk17/create-image-from-git-tag.Jenkinsfile:14]: gitRepoURL, getCommitSha and updateGithubCommitStatus I think are not needed here

File: maven/3.8.7/openjdk17/create-image-from-git-tag.Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            }
+        }
+        stage('Remove Unused docker image') {
+            steps{
+                sh "docker rmi ${imageName}:${params.TAG}"
+            }
+        }
+    }
+    post {
+        failure {
```
Comment (agentili) [inline maven/3.8.7/openjdk17/create-image-from-git-tag.Jenkinsfile:91]: Is this failure step actually needed.
"env.BRANCH_NAME" I don't know if it is present as a variable

File: maven/3.8.7/openjdk17/create-image-from-git-tag.Jenkinsfile [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+            emailext( attachLog: true,
+                compressLog: true,
+                recipientProviders: [culprits()],
+                subject: "[${currentBuild.projectName}] Failed Pipeline: ${currentBuild.fullDisplayName}",
+                body: "Something is wrong with ${env.BUILD_URL} on branch ${env.BRANCH_NAME}, please check the log"
+            )
+        }
+        always {
+            updateGithubCommitStatus("Jenkins Job ${env.BUILD_NUMBER} - Result: ${currentBuild.currentResult} ")
```
Comment (agentili) [inline maven/3.8.7/openjdk17/create-image-from-git-tag.Jenkinsfile:100]: is it actually usefull here when we build from TAG?

File: maven/3.8.7/openjdk17/create-image-from-git-tag.Jenkinsfile [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+            sh "export BUILD_ID=$BUILD_ID"
```
Comment (agentili) [inline maven/3.8.7/openjdk17/create-image-from-git-tag.Jenkinsfile:101]: what this meant for?

File: maven/3.8.7/openjdk17/mvn-entrypoint.sh [hunk 8]
```diff
@@ -0,0 +1,50 @@
+#! /bin/sh -eu
+
+# Copy files from /usr/share/maven/ref into ${MAVEN_CONFIG}
+# So the initial ~/.m2 is set with expected content.
+# Don't override, as this is just a reference setup
+
+copy_reference_files() {
```
Comment (agentili) [inline maven/3.8.7/openjdk17/mvn-entrypoint.sh:7]: what this script is actually doing?

Comment (agentili) [reply maven/3.8.7/openjdk17/mvn-entrypoint.sh:7]: Clarified

Comment (AlessandroDeMartini) [reply maven/3.8.7/openjdk17/mvn-entrypoint.sh:7]: I believe this is used to set up all the mvn environment as we are used to. I was inspired here 

https://github.com/carlossg/docker-maven/tree/32760466401d45fabb8166b5dcf9003b07598918/amazoncorretto-17

File: maven/3.8.7/openjdk17/create-image-from-git-tag.Jenkinsfile (hunk 4 above)
Comment (AlessandroDeMartini) [reply maven/3.8.7/openjdk17/create-image-from-git-tag.Jenkinsfile:14]: done

File: maven/3.8.7/openjdk17/create-image-from-git-tag.Jenkinsfile (hunk 6 above)
Comment (AlessandroDeMartini) [reply maven/3.8.7/openjdk17/create-image-from-git-tag.Jenkinsfile:100]: removed

File: maven/3.8.7/openjdk17/create-image-from-git-tag.Jenkinsfile (hunk 7 above)
Comment (AlessandroDeMartini) [reply maven/3.8.7/openjdk17/create-image-from-git-tag.Jenkinsfile:101]: this is used to clean the environment. I removed it

File: maven/3.8.7/openjdk17/create-image-from-git-tag.Jenkinsfile (hunk 5 above)
Comment (AlessandroDeMartini) [reply maven/3.8.7/openjdk17/create-image-from-git-tag.Jenkinsfile:61]: edited with params.TAG

**Review by aielloaldo — APPROVED** [review]

File: openjdk/17/debian/create-image-from-git-tag.Jenkinsfile (hunk 1 above)
Comment (AlessandroDeMartini) [reply openjdk/17/debian/create-image-from-git-tag.Jenkinsfile:61]: edited with params.TAG

**Pull request merged**